pkg/storage/etcd3/event.go

```diff
@@ -30,14 +30,16 @@ type event struct {
 	isCreated bool
 }
 
+// parseKV converts a KeyValue and previous value to a synthetic event.
+// isCreated is true if kv.ModRevision == kv.CreateRevision OR if prevVal is nil.
 func parseKV(kv *mvccpb.KeyValue, prevVal []byte) *event {
 	return &event{
 		key:       string(kv.Key),
 		value:     kv.Value,
 		prevValue: prevVal,
 		rev:       kv.ModRevision,
 		isDeleted: false,
-		isCreated: kv.ModRevision == kv.CreateRevision,
+		isCreated: kv.ModRevision == kv.CreateRevision || prevVal == nil,
```

Comment:
Not that it matters for this PR necessarily, but a non-nil, zero-byte prevVal would show up as isCreated = false. You could check for len=0 instead if you wanted.